4단계 - @Configuration 설정(힌트)

catsbi opened the pull request:
안녕하세요 리뷰어님. 4단계에서는 미비했던 테스트 케이스를 작성하는데 집중해봤습니다. 
늘 리뷰해주시는것에 감사합니다. 👍 

이번 리뷰도 잘 부탁드립니다!

catsbi committed:
feat: BeanRegister 구현체 테스트 케이스 작성

catsbi committed:
feat: BeanScanner 구현체 테스트 케이스 작성

kang-hyungu commented on src/test/java/core/di/factory/ClassBeanRegisterTest.java:
```diff
@@ -0,0 +1,74 @@
+
+    @DisplayName("getParameterCount 메서드는 생성자 매개변수의 갯수를 반환한다.")
+    @Test
+    void itIsReturnsParameterCountOfConstructor() {
```
이 테스트가 실패하네요.
register 객체에서 생성자 정보를 가져오는 작업이 필요해보입니다.

catsbi replied:
아 이 부분을 놓쳤네요 ㅠㅠ 제출하기전에 다 통과하는걸로 봤는데 왜 이부분을 놓쳤지..

kang-hyungu commented on src/test/java/core/di/factory/ClasspathBeanScannerTest.java:
```diff
@@ -0,0 +1,30 @@
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class ClasspathBeanScannerTest {
```
ClasspathBeanScanner 클래스를 보니 scan 메서드는 테스트 가능할 것 같네요.
테스트를 추가해서 정상 동작하는지 검증해보면 어떨까요?

catsbi replied:
좋은 생각이십니다. 관성적으로 클래스의 컴포넌트스캔 애노테이션만 신경쓰다보니 놓쳤네요!

kang-hyungu reviewed (changes requested):
한솔님 3단계에서 잘 진행해주셔서 4단계는 크게 수정할 부분은 없네요.
테스트 코드만 더 추가해서 의도한대로 정상 동작하는지 점검해보면 좋을 것 같습니다.

lhs0610 committed:
refactor: 피드백 반영

- 실패하는 테스트(itIsReturnsParameterCountOfConstructor) 수정
- 문자열기반의 scan 메서드 테스트 케이스 추가

catsbi commented:
놓친부분들 피드백 반영했습니다. 매번 나름 신경쓴다고 하는데 놓치는부분들이 생기네요. 좀 더 챙기도록 하겠습니다!

kang-hyungu reviewed (approved):
한솔님 피드백 반영 잘 해주셨습니다.
pr은 머지 처리할게요. 즐거운 추석 보내세요.
